lib/pre_assembly/digital_object.rb

```diff
@@ -277,8 +277,8 @@ def register_in_dor(params)
           all_pids=sourceid_pids << @pid
           all_pids.each do |pid|
             begin
+              Dor::SearchService.solr.delete_by_id(pid)  # should be unnecessary, but handles an edge case where the object is not in Fedora, but is in Solr
```

Comment:
good question, the lines used to be reversed since sometimes deleting from Fedora would not immediately delete from Solr, and then the register would fail again (since there would be a duped sourceid).  Yesterday I ran into the problem of not in Fedora, but in Solr, which is why I reversed the line.  In actuality, this is a rare scenario.  Basically it is triggered by a crashed or forced quit pre-assembly run where the object was registered but the main action on the object never completed.  When you resume the pre-assembly run, the code attempts to do that object over again but fails on register since the register actually succeeded and thus the source ID already exists.  This was an attempt to perform some auto-cleanup.  You can still recover by manually deleting the object from Fedora and/or Solr as needed (which is what i did yesterday).